### PR TITLE
Delete Procfile

### PR DESCRIPTION
**Pull request by rebkwok**

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -139,6 +139,12 @@ COPY . /app
 # are readable by all.
 COPY --from=node-builder /usr/src/app/assets/dist /app/assets/dist
 
+# set up the django app
+RUN SECRET_KEY=dummy-key python /app/manage.py migrate
+RUN SECRET_KEY=dummy-key python /app/manage.py ensure_groups
+RUN SECRET_KEY=dummy-key python /app/manage.py collectstatic --no-input
+RUN SECRET_KEY=dummy-key python /app/manage.py createcachetable
```
Comment (rebkwok) [inline docker/Dockerfile:146]: @bloodearnest this is the only env variable that needs to be set, and this was the only way I could seem to get it to work.  Not sure if there is a better way

File: output_explorer/settings.py [hunk 2]
```diff
@@ -193,9 +193,9 @@
 )
 
 SOCIAL_AUTH_ADMIN_USER_SEARCH_FIELDS = ["username", "first_name", "email"]
-SOCIAL_AUTH_NHSID_KEY = env.str("SOCIAL_AUTH_NHSID_KEY")
-SOCIAL_AUTH_NHSID_SECRET = env.str("SOCIAL_AUTH_NHSID_SECRET")
-SOCIAL_AUTH_NHSID_API_URL = env.str("SOCIAL_AUTH_NHSID_API_URL")
+SOCIAL_AUTH_NHSID_KEY = env.str("SOCIAL_AUTH_NHSID_KEY", default="")
+SOCIAL_AUTH_NHSID_SECRET = env.str("SOCIAL_AUTH_NHSID_SECRET", default="")
+SOCIAL_AUTH_NHSID_API_URL = env.str("SOCIAL_AUTH_NHSID_API_URL", default="")
```
Comment (rebkwok) [inline output_explorer/settings.py:198]: Given that the NHS ID stuff is not used and currently not working, I don't think it's terrible to set defaults for these variables so we don't need them for the manage commands that run in the build

File: docker/Dockerfile (hunk 1 above)
Comment (bloodearnest) [reply docker/Dockerfile:146]: This is great.

But I think `migrate`, `ensure_groups` and possilby `createcachetable` still need to be run in the entrypoint? As I think their changes are persisted to the db, which is not in the docker image?

File: output_explorer/settings.py (hunk 2 above)
Comment (bloodearnest) [reply output_explorer/settings.py:198]: Yep, seems reasonable

File: docker/Dockerfile (hunk 1 above)
Comment (rebkwok) [reply docker/Dockerfile:146]: Oh yes, you're right, that hadn't occurred to me.  I'll move them back.

**Comment by bloodearnest** [comment]
Should we remove the collectstatic from the prod entrypoint script too?

**Comment by rebkwok** [comment]
> Should we remove the collectstatic from the prod entrypoint script too?

I think I already did?

**Review by bloodearnest — APPROVED** [review]

**Pull request merged**